### PR TITLE
[fix](compile) fix macos compile error

### PR DESCRIPTION
**Pull request by Tech-Circle-48**
## Proposed changes

Issue Number: close #xxx

doris/be/src/vec/functions/functions_logical.cpp:23:10: fatal error: 'bits/ranges_algo.h' file not found in macOS

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/analysis/DateLiteral.java [hunk 1]
```diff
@@ -59,11 +59,7 @@
 import java.time.temporal.IsoFields;
 import java.time.temporal.TemporalAccessor;
 import java.time.temporal.WeekFields;
-import java.util.Collections;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
-import java.util.TimeZone;
+import java.util.*;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/DateLiteral.java:62]: revert this change

Comment (Tech-Circle-48) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/DateLiteral.java]: it's done

**Comment by mrhhsg** [comment]
run buildall

**Comment by Tech-Circle-48** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

**Comment by Tech-Circle-48** [comment]
run buildall

**Comment by Tech-Circle-48** [comment]
run buildall

**Review by adonis0147 — DISMISSED** [review]

**Comment by adonis0147** [comment]
run buildall

**Comment by Tech-Circle-48** [comment]
run buildall

**Comment by adonis0147** [comment]
Hi @Tech-Circle-48 , the `BE UT (macOS)` failed. 
```shell
/Users/runner/work/doris/doris/be/src/vec/functions/functions_logical.cpp:23:10: fatal error: 'bits/ranges_algo.h' file not found
#include <bits/ranges_algo.h>
         ^~~~~~~~~~~~~~~~~~~~
```
The header should be replaced by `<algorithm>` + `<ranges>`.

**Comment by Tech-Circle-48** [comment]
run buildall

**Review by Yukang-Lian — APPROVED** [review]
LGTM

**Comment by adonis0147** [comment]
run p0

**Comment by adonis0147** [comment]
run cloud_p1

**Review by adonis0147 — APPROVED** [review]
